app/src/main/java/com/randomappsinc/studentpicker/grouping/GroupMakingSettings.java

```diff
@@ -1,37 +1,39 @@
 package com.randomappsinc.studentpicker.grouping;
 
-class GroupMakingSettings {
+public class GroupMakingSettings {
 
     private static final int DEFAULT_NAMES_PER_GROUP = 2;
 
     private int nameListSize;
     private int numOfNamesPerGroup;
     private int numOfGroups;
 
-    GroupMakingSettings(int nameListSize) {
+    public GroupMakingSettings(int nameListSize) {
         this.nameListSize = nameListSize;
         this.numOfNamesPerGroup = DEFAULT_NAMES_PER_GROUP;
-
-        int numGroupsToFill = nameListSize / DEFAULT_NAMES_PER_GROUP;
-        numOfGroups = (nameListSize % DEFAULT_NAMES_PER_GROUP) > 0
-                ? numGroupsToFill + 1
-                : numGroupsToFill;
     }
 
-    int getNumOfNamesPerGroup() {
+    public int getNumOfNamesPerGroup() {
         return numOfNamesPerGroup;
     }
 
-    void setNumOfNamesPerGroup(int numNamesPerGroup) {
+    public void setNumOfNamesPerGroup(int numNamesPerGroup) {
         this.numOfNamesPerGroup = numNamesPerGroup;
     }
 
-    int getNumOfGroups() {
+    public int getNumOfGroups() {
         return numOfGroups;
     }
 
-    void setNumOfGroups(int numOfGroups) {
-        this.numOfGroups = numOfGroups;
+    public void setNumOfGroups(int numOfGroups) {
+        if (numOfGroups != 0) {
+            this.numOfGroups = numOfGroups;
+        } else {
+            int numGroupsToFill = nameListSize / DEFAULT_NAMES_PER_GROUP;
```

Comment:
I feel like we're overthinking the setter. If the user is purposely setting the value to 0, I think you should just let them do it. This code makes this API less readable and introduces complication since the 0 mutation should be reflected in the UI as well. you should just let my old code die, lol.

Reply:
if I deleted this code the default value will always be 1 and we will not be able to set the initial value of groups number as names list size